### PR TITLE
[5.2] Feed Modules

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
This PR updates both the site and the admin feed modules.

The original intent was just to remove the inline styles but it became clear that neither of these modules were fit for purpose.

- inline style replaced with html valid and accessible dir elements
- noopener noreferrer
- unused variables removed
- adopted consistent coding of the if statements as used in all other tmpl (thanks @hackware)
- kept the existing site module css classes (even though they dont exist in cassiopeia for b/c with 3pd templates)
- improve xss protection on admin module links
- add some basic styling on admin module
- remove overly complex and superflous rtl/ltr code

### Testing Instructions
Create 3 newsfeed modules for both site and admin as below

1. set the direction to rtl and the url to `https://www.aljazeera.net/aljazeerarss/a7c186be-1baa-4bd4-9d80-a84db769f779/73d0e1b4-532f-45ef-b135-bfdff8b8cab9`

2. Leave direction at the default ltr and the url to https://www.joomla.org/announcements.feed?type=rss

3. set the url to an invalid rss feed eg https://www.joomla.org

## Before PR

image

image

## After PR
no visual changes in cassiopeia

image

### Additional Comment
The Component also needs some similar reworking

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

File: administrator/modules/mod_feed/tmpl/default.php [hunk 1]
```diff
@@ -14,105 +14,75 @@
...
+                            <?php
+                                // Strip the images.
+                                $text = OutputFilter::stripImages($text);
+                                $text = HTMLHelper::_('string.truncate', $text, $params->get('word_count', 0));
+                                echo str_replace('&apos;', "'", $text);
+                            ?>
+                            </div>
+                        <?php endif; ?>
+                    </li>
+                <?php } ?>
```
Comment (Quy) [inline administrator/modules/mod_feed/tmpl/default.php]: Why not keep it as `endfor`?

Comment (brianteeman) [reply administrator/modules/mod_feed/tmpl/default.php]: because it is not needed

Comment (Quy) [reply administrator/modules/mod_feed/tmpl/default.php]: But it is more readable for template designers just as `endif`.

Comment (HLeithner) [reply administrator/modules/mod_feed/tmpl/default.php]: we use alternative PHP syntax for template files that's the reason why we us if/endif and for/endfor, sadly there is no code style for html/php mix but doesn't mean we have not to keep our weak standards for this topic.

File: administrator/modules/mod_feed/tmpl/default.php [hunk 2]
```diff
@@ -14,105 +14,75 @@
 use Joomla\CMS\Language\Text;
 use Joomla\Filter\OutputFilter;
 
-// Check if feed URL has been set
-if (empty($rssurl)) {
-    echo '<div>' . Text::_('MOD_FEED_ERR_NO_URL') . '</div>';
```
Comment (Quy) [inline administrator/modules/mod_feed/tmpl/default.php:19]: Deprecate/remove language strings.

File: administrator/modules/mod_feed/tmpl/default.php [hunk 3]
```diff
@@ -14,105 +14,75 @@
...
```
Comment (Quy) [inline administrator/modules/mod_feed/tmpl/default.php]:
```suggestion
```

File: modules/mod_feed/tmpl/default.php [hunk 4]
```diff
@@ -14,107 +14,75 @@
 use Joomla\CMS\Language\Text;
 use Joomla\Filter\OutputFilter;
 
-// Check if feed URL has been set
-if (empty($rssurl)) {
-    echo '<div>' . Text::_('MOD_FEED_ERR_NO_URL') . '</div>';
```
Comment (Quy) [inline modules/mod_feed/tmpl/default.php]:
```suggestion
```

**Comment by ufuk-avcu** [comment]
I have tested this item :white_check_mark: successfully on da9a38015058d7b0d9b4ca5961fc8dbceb6b2d92<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42000">issues.joomla.org/tracker/joomla-cms/42000</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on da9a38015058d7b0d9b4ca5961fc8dbceb6b2d92<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42000">issues.joomla.org/tracker/joomla-cms/42000</a>.</sub>

**Comment by hl-dev-DE** [comment]
I have tested this item :white_check_mark: successfully on da9a38015058d7b0d9b4ca5961fc8dbceb6b2d92<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42000">issues.joomla.org/tracker/joomla-cms/42000</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42000">issues.joomla.org/tracker/joomla-cms/42000</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.2-dev.

**Comment by brianteeman** [comment]
Closed due to lack of any interest - automatic rebases break things.

**Pull request closed without merging**